plugin/advisor/pg/advisor_naming_unique_key_convention.go

```diff
@@ -0,0 +1,180 @@
+func (check *NamingUKConventionAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	root, errAdvice := parseStatement(statement)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySQLReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+
+	format, templateList, maxLength, err := advisor.UnmarshalNamingRulePayloadAsTemplate(ctx.Rule.Type, ctx.Rule.Payload)
+	if err != nil {
+		return nil, err
+	}
+
+	checker := &namingUKConventionChecker{
+		level:        level,
+		title:        string(ctx.Rule.Type),
+		format:       format,
+		maxLength:    maxLength,
+		templateList: templateList,
+		catalog:      ctx.Catalog,
+	}
+
+	for _, stmtNode := range root {
+		ast.Walk(checker, stmtNode)
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    advisor.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+
+	return checker.adviceList, nil
+}
+
+type namingUKConventionChecker struct {
+	adviceList   []advisor.Advice
+	level        advisor.Status
+	title        string
+	format       string
+	maxLength    int
+	templateList []string
+	catalog      catalog.Catalog
+}
+
+// Visit implements ast.Visitor interface
+func (checker *namingUKConventionChecker) Visit(in ast.Node) ast.Visitor {
+	indexDataList := checker.getMetaDataList(in)
+
+	for _, indexData := range indexDataList {
+		regex, err := getTemplateRegexp(checker.format, checker.templateList, indexData.metaData)
+		if err != nil {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    advisor.Internal,
+				Title:   "Internal error for unique key naming convention rule",
+				Content: fmt.Sprintf("%q meet internal error %q", in.Text(), err.Error()),
+			})
+			continue
+		}
+		if !regex.MatchString(indexData.indexName) {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    advisor.NamingUKConventionMismatch,
+				Title:   checker.title,
+				Content: fmt.Sprintf(`Unique key in table "%s" mismatches the naming convention, expect %q but found "%s"`, indexData.tableName, regex, indexData.indexName),
+			})
+		}
+		if checker.maxLength > 0 && len(indexData.indexName) > checker.maxLength {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    advisor.NamingUKConventionMismatch,
+				Title:   checker.title,
+				Content: fmt.Sprintf(`Unique key "%s" in table "%s" mismatches the naming convention, its length should be within %d characters`, indexData.indexName, indexData.tableName, checker.maxLength),
+			})
+		}
+	}
+
+	return checker
+}
+
+// getMetaDataList returns the list of foreign key with metadata.
+func (checker *namingUKConventionChecker) getMetaDataList(in ast.Node) []*indexMetaData {
+	var res []*indexMetaData
+	switch node := in.(type) {
```

Comment:
We should consider the `RenameConstraint` and `RenameIndex` statements.

Reply:
ditto

Since we don't have the key list here, how can we know whether the renamed uk name is good? If the convention produce a fixed name, any rename can be invalid.

Reply:
Oops. Catalogs should help to find the index column name, let me take a closer look.

Reply:
Yes, you can get the index definition from the catalog.